add-on/panels/js/roomViews.jsx

```diff
@@ -412,8 +412,14 @@ loop.roomViews = (function(mozL10n) {
      */
     leaveRoom: function() {
       if (this.state.used) {
+        // The room has be used, so we might want to display the feedback view.
+        // Therefore we leave the room to ensure that we have stopped sharing etc,
+        // then trigger the call terminated handler that'll do the right thing
+        // for the feedback view.
         this.props.dispatcher.dispatch(new sharedActions.LeaveRoom());
+        this.props.onCallTerminated();
```

Comment:
This is certainly an improvement from what was there before.  

That said, it feels odd that the closeWindow is hidden behind the onCallTerminated in one of the if clauses but not the other.

This combined with the fact that closeWindow implicitly causes an unload event to be fired, which then is handle and causes a WindowClose action to be dispatched, makes our window shutdown code fairly convoluted and hard to follow without a bunch of work.  I'm not sure if there's anything to be done about that here or not, though.  Maybe onCallTerminated (and handleCallTerminated that implements it) should be named in a way to describe what it does, rather than when it executes?